servicetalk-grpc-netty/src/test/java/io/servicetalk/grpc/netty/GrpcMessageEncodingTest.java

```diff
@@ -213,103 +213,95 @@ public StreamingHttpServiceFilter create(final StreamingHttpService service) {
 
-                        final List<String> expectedReqAcceptedEncodings = (clientSupportedEncodings == null) ?
-                                        emptyList() :
-                                        clientSupportedEncodings.stream()
-                                                .filter((enc) -> enc != identity())
-                                                .map((ContentCodec::name))
-                                                .map((CharSequence::toString))
-                                                .collect(toList());
+                        final List<String> expectedReqAcceptedEncodings = encodingsAsStrings(clientSupportedEncodings);
 
                         assertTrue("Request encoding should be present in the request headers",
                                 contentEquals(reqEncoding.name(), request.headers().get(MESSAGE_ENCODING, "identity")));
                         if (!expectedReqAcceptedEncodings.isEmpty() && !actualReqAcceptedEncodings.isEmpty()) {
                             assertEquals(expectedReqAcceptedEncodings, actualReqAcceptedEncodings);
                         }
                     } catch (Throwable t) {
-                        t.printStackTrace();
+                        errors.add(t);
                         throw t;
                     }
 
                     return super.handle(ctx, request, responseFactory).map((response -> {
                         try {
-                            final List<String> actualRespAcceptedEncodings = stream(response.headers()
-                                    .get(MESSAGE_ACCEPT_ENCODING, "NOT_PRESENT").toString().split(","))
-                                    .map((String::trim)).collect(toList());
+                            _handle(clientSupportedEncodings, serverSupportedEncodings, response);
+                        } catch (Throwable t) {
+                            errors.add(t);
+                            throw t;
+                        }
 
-                            final List<String> expectedRespAcceptedEncodings = (serverSupportedEncodings == null) ?
-                                    emptyList() :
-                                    serverSupportedEncodings.stream()
-                                            .filter((enc) -> enc != identity())
-                                            .map((ContentCodec::name))
-                                            .map((CharSequence::toString))
-                                            .collect(toList());
+                        return response;
+                    }));
+                }
 
-                            if (!expectedRespAcceptedEncodings.isEmpty() && !actualRespAcceptedEncodings.isEmpty()) {
-                                assertEquals(expectedRespAcceptedEncodings, actualRespAcceptedEncodings);
-                            }
+                private void _handle(final List<ContentCodec> clientSupportedEncodings,
```

Comment:
We usually use `handle0` pattern for internal methods instead of the leading underscore